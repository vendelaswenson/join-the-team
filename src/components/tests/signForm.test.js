import React from 'react'
import { render, fireEvent, waitFor } from '@testing-library/react'
import SignForm from '../SignForm'

const mockOnFormSubmit = jest.fn()

const renderComponent = () => {
  return render(<SignForm onFormSubmit={mockOnFormSubmit} />)
}

it('renders without crashing', async () => {
  const utils = render(<SignForm />)
  expect(utils).toMatchSnapshot()
  expect(utils).toBeTruthy()
})

describe('SignForm', () => {
  it('renders the form fields and submit button', () => {
    const { getByPlaceholderText, getByText } = renderComponent()

    expect(getByPlaceholderText('Name')).toBeTruthy()
    expect(getByPlaceholderText('Email')).toBeTruthy()
    expect(getByText("I'm in, sign me up!")).toBeTruthy()
  })

  it('submits the form with the provided data', async () => {
    const { getByPlaceholderText, getByText } = renderComponent()

    fireEvent.change(getByPlaceholderText('Name'), {
      target: { value: 'John Doe' },
    })

    fireEvent.change(getByPlaceholderText('Email'), {
      target: { value: 'john@example.com' },
    })

    fireEvent.click(getByText("I'm in, sign me up!"))

    await waitFor(() => {
      expect(mockOnFormSubmit).toHaveBeenCalledWith('John Doe')
    })
  })
})
