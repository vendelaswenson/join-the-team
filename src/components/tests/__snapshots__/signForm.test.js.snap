// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <h3
          class="navy f4"
        >
          Team player - Be positive - Beat yesterday
        </h3>
        <p
          class="measure black-60 lh-copy f6"
        >
          Together we re-define the experience of online gaming through gamification and novel technical solutions.
        </p>
        <div
          class="input--container"
        >
          <input
            class="input--field"
            name="firstName"
            placeholder="Name"
          />
          <input
            class="input--field"
            name="email"
            placeholder="Email"
          />
        </div>
        <div
          class="mv3"
        >
          <input
            id="terms"
            name="terms"
            type="checkbox"
          />
          <label
            for="terms"
          >
            I agree to the terms
            <span
              class="invisible--text white f7"
            >
              and I'll bring nice fika every friday ;)
            </span>
             
          </label>
        </div>
        <button
          class="form--button ph4 pv2 bg-blue white br2 bn mt3 grow pointer"
          data-ga="submit"
          type="submit"
        >
          I'm in, sign me up!
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <h3
        class="navy f4"
      >
        Team player - Be positive - Beat yesterday
      </h3>
      <p
        class="measure black-60 lh-copy f6"
      >
        Together we re-define the experience of online gaming through gamification and novel technical solutions.
      </p>
      <div
        class="input--container"
      >
        <input
          class="input--field"
          name="firstName"
          placeholder="Name"
        />
        <input
          class="input--field"
          name="email"
          placeholder="Email"
        />
      </div>
      <div
        class="mv3"
      >
        <input
          id="terms"
          name="terms"
          type="checkbox"
        />
        <label
          for="terms"
        >
          I agree to the terms
          <span
            class="invisible--text white f7"
          >
            and I'll bring nice fika every friday ;)
          </span>
           
        </label>
      </div>
      <button
        class="form--button ph4 pv2 bg-blue white br2 bn mt3 grow pointer"
        data-ga="submit"
        type="submit"
      >
        I'm in, sign me up!
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
